services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/DefaultRulesSpec.groovy

```diff
@@ -219,6 +293,83 @@ class DefaultRulesSpec extends HealthCheckSpecification {
+            missingRules == deletedRules
+            excessRules.empty
+            properRules.sort() == sw.defaultCookies.findAll { it != data.cookie }.sort()
+        }
+        verifyAll(northbound.validateSwitch(sw.dpId)) {
+            rules.missing == deletedRules
+            rules.misconfigured.empty
+            rules.excess.empty
+            rules.proper.sort() == sw.defaultCookies.findAll { it != data.cookie }.sort()
+        }
+
+        and: "Install default rules back"
+        multitableIsEnabled ?: northbound.updateSwitchProperties(sw.dpId, initSwitchProperty)
+        northbound.installSwitchRules(sw.dpId, InstallRulesAction.INSTALL_DEFAULTS)
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.getSwitchRules(sw.dpId).flowEntries.size() == defaultRules.size()
+        }
+
+        where:
+        [data, sw] << [
+                [
+                        [
+                                deleteRulesAction: DeleteRulesAction.REMOVE_MULTITABLE_PRE_INGRESS_PASS_THROUGH,
+                                cookie           : Cookie.MULTITABLE_PRE_INGRESS_PASS_THROUGH_COOKIE
+                        ],
+                        [
+                                deleteRulesAction: DeleteRulesAction.REMOVE_MULTITABLE_INGRESS_DROP,
+                                cookie           : Cookie.MULTITABLE_INGRESS_DROP_COOKIE
+                        ],
+                        [
+                                deleteRulesAction: DeleteRulesAction.REMOVE_MULTITABLE_POST_INGRESS_DROP,
+                                cookie           : Cookie.MULTITABLE_POST_INGRESS_DROP_COOKIE
+                        ],
+                        [
+                                deleteRulesAction: DeleteRulesAction.REMOVE_MULTITABLE_EGRESS_PASS_THROUGH,
+                                cookie           : Cookie.MULTITABLE_EGRESS_PASS_THROUGH_COOKIE
+                        ],
+                        [
+                                deleteRulesAction: DeleteRulesAction.REMOVE_MULTITABLE_TRANSIT_DROP,
+                                cookie           : Cookie.MULTITABLE_TRANSIT_DROP_COOKIE
+                        ]
+                ],
+                getTopology().getActiveSwitches().unique { activeSw -> activeSw.description }
+        ].combinations().findAll { dataPiece, theSw ->
+            //OF_12 switches has only one broadcast rule, so not all iterations will be applicable
+            !(theSw.ofVersion == "OF_12" && dataPiece.cookie != Cookie.VERIFICATION_BROADCAST_RULE_COOKIE) &&
+                    //dropping this rule on WB5164 will lead to disco-packet storm. Reason: #2595
+                    !(theSw.wb5164 && dataPiece.cookie == Cookie.DROP_VERIFICATION_LOOP_RULE_COOKIE)
+        }
+    }
+
     void compareRules(actualRules, expectedRules) {
         assert expect(actualRules.sort { it.cookie }, sameBeanAs(expectedRules.sort { it.cookie })
                 .ignoring("byteCount")
```

Comment:
Are huge amount of important tests are missing.
Switch validation and synchronization is absolutely not covered. This will be the most important operation to happen in prod env. Imagine the migration process. How the switch will behave itself during sync? How the default rules will behave and how the flow rules will behave?

Reply:
switch validation is covered in the first test

Reply:
test was added for that purpose (b38cdc65973b5e0e4aac49533cba62eb56bc7266)